src/components/withWindowDimensions/index.tsx

```diff
@@ -43,7 +40,7 @@ function WindowDimensionsProvider(props) {
     });
 
     useEffect(() => {
-        const onDimensionChange = (newDimensions) => {
+        const onDimensionChange = (newDimensions: {window: ScaledSize}) => {
```

Comment:
maybe worth to define newDimension type, looks like You used it twice

Reply:
Great idea, done!